### ISSUE TITLE
Unexpected type error because missing capture conversion

### ISSUE THREAD

**Issue by CharlesZ-Chen** [typetools/checker-framework#759]
Take:

```java
class IncompatibleTypes {
    void possibleValues(final Class<? extends Enum> enumType) {
        lowercase(enumType.getEnumConstants());
    }

    <T extends Enum<T>> void lowercase(final T... items) {
    }
}
```

Running any checker on this file will result in:

```
error: [argument.type.incompatible] incompatible types in argument.
        lowercase(enumType.getEnumConstants());
                                           ^
  found   : ?[ extends @UnknownKeyFor Enum super @KeyForBottom Void] @UnknownKeyFor []
  required: ?[ extends @UnknownKeyFor Enum super @KeyForBottom Void] @UnknownKeyFor []
```

The wildcard should be captured properly and no error is expected here.

(The following three checkers produce different errors which should also be investigated:

```
GUI Effect Checker (no imcompatible error, passed)
Lock Checker
	found : ? extends @GuardedBy @LockPossiblyHeld Enum @GuardedBy @LockPossiblyHeld []
	 required: @LockPossiblyHeld ? extends @GuardedBy @LockPossiblyHeld Enum @GuardedBy @LockPossiblyHeld []
Aliasing Checker
	found : ? extends @MaybeAliased @MaybeLeaked Enum @MaybeAliased @MaybeLeaked []
	required: @MaybeLeaked ? extends @MaybeAliased @MaybeLeaked Enum @MaybeAliased @MaybeLeaked []
```
)

**Comment by smillst** [comment]
Thanks for reporting this issue.  Your test case issues javac warnings:
````
javacheck -processor interning Issue759.java -Xlint:unchecked
Issue759.java:10: warning: [unchecked] unchecked method invocation: method lowercase in class IncompatibleTypes is applied to given types
        lowercase(enumType.getEnumConstants());
                 ^
  required: T[]
  found: CAP#1[]
  where T is a type-variable:
    T extends Enum<T> declared in method <T>lowercase(T...)
  where CAP#1 is a fresh type-variable:
    CAP#1 extends Enum from capture of ? extends Enum
````
Generally if javac issues an unchecked warning, the Checker Framework is likely to issue a type-checking error at the same location. However, the test case below, which is similar to your test case, also issues an unexpected type-checker error.  I've added the test case to the repository.
````
class IncompatibleTypes {
    void possibleValues(final Class<? extends Enum> enumType) {
        lowercase(enumType.getEnumConstants());
    }
    <T> void lowercase(final T items) {
    }
}
````

**Comment by wmdietl** [comment]
I don't think anything here is special to enums. One could write a similar example without using enums.

**Comment by smillst** [comment]
>I don't think anything here is special to enums. One could write a similar example without using enums.

I think it has something to do with this comment:
https://github.com/typetools/checker-framework/blob/master/framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java#L871-L877

Also, if you wrote the same example without using Enum, then it would be a raw type.

**Comment by wmdietl** [comment]
Test case without enum:

```
class Gen<T extends Gen<T>> {
    T[] getConstants() { return null; }
}

class IncompatibleTypes {
    void possibleValues(final Gen<?> genType) {
        lowercase(genType.getConstants());
    }

    <S> void lowercase(final S items) {
    }
}
```

No raw type involved.

**Comment by smillst** [comment]
Thanks for the test case!
